src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -162,7 +162,7 @@ export function run(
   const casesPerBatch = (function () {
     switch (cfg.inputSource) {
       case 'const':
-        return 256; // Arbitrary limit, to ensure shaders aren't too large
+        return 64; // Arbitrary limit, to ensure shaders aren't too large
```

Comment:
github won't let me comment lower down.

Looking at this again, we do:
```
pushErrorScope('validation');
submit all the batches
popErrorScope()
checkResults.forEach(f => f());
```
maybe instead we should do:
```
promises = [];
for each batch:
  pushErrorScope
  checkBatch = submitBatch
  promises.push(popErrorScope().then(checkBatch));
  
await Promise.all(promises);
```

the reason is that waiting for popErrorScope after ALL the batches will wait for them all to finish validating

Reply:
**wait for them all to run through validation in Dawn. the `f` that's returned by `submitBatch` will still run one-by-one with heartbeats in between in the current code.

Reply:
Done